Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -112,61 +115,84 @@ namespace Multiplayer
-        m_drawParams.m_position = AZ::Vector3(static_cast<float>(viewportSize.m_width), static_cast<float>(viewportSize.m_height), 1.0f) + AZ::Vector3(viewportConnectionBottomRightBorderPadding) * viewport->GetDpiScalingFactor();
-        
+        AZ::Color connectionStateColor;
         switch (connectionState)
         {
         case AzNetworking::ConnectionState::Connecting:
-            m_drawParams.m_color = AZ::Colors::Yellow;
+            connectionStateColor = AZ::Colors::Yellow;
             break;
         case AzNetworking::ConnectionState::Connected:
-            m_drawParams.m_color = AZ::Colors::Green;
+            connectionStateColor = AZ::Colors::Green;
             break;
         case AzNetworking::ConnectionState::Disconnecting:
-            m_drawParams.m_color = AZ::Colors::Orange;
+            connectionStateColor = AZ::Colors::Yellow;
             break;
         case AzNetworking::ConnectionState::Disconnected:
-            m_drawParams.m_color = AZ::Colors::Red;
+            connectionStateColor = AZ::Colors::Red;
             break;
-        default: m_drawParams.m_color = AZ::Colors::White;
+        default:
+            connectionStateColor = AZ::Colors::White;
+        }
+
+        // Draw our host's remote ip address
+        const auto multiplayerSystemComponent = AZ::Interface<IMultiplayer>::Get();
+        MultiplayerAgentType agentType = multiplayerSystemComponent->GetAgentType();
+        if (agentType == MultiplayerAgentType::Client)
+        {
+            auto hostAddressText = AZStd::fixed_string<32>::format("Server IP %s", hostIpAddress.GetString().c_str());
+            DrawConnectionStatusLine(hostAddressText.c_str(), connectionStateColor);
         }
 
-        m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, connectionStateText);
+        // Draw the connect state (example: Connected or Disconnected)
+        DrawConnectionStatusLine(ToString(connectionState).data(), connectionStateColor);
+
+        // Draw the status title
+        DrawConnectionStatusLine(ClientStatusTitle, AZ::Colors::White);
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::DrawConnectionStatusLine(const char* line, const AZ::Color& color)
+    {
+        m_drawParams.m_color = color;
+        m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, line);
 
-        // Draw the status title above the current status
-        const float textHeight = m_fontDrawInterface->GetTextSize(m_drawParams, connectionStateText).GetY();
-        m_drawParams.m_color = AZ::Colors::White;
+        // Status text renders in the lower right corner, so we draw from the bottom up.
+        // Move the font draw position up to get ready for the next text line.
+        const float textHeight = m_fontDrawInterface->GetTextSize(m_drawParams, line).GetY();
         m_drawParams.m_position.SetY(m_drawParams.m_position.GetY() - textHeight - m_lineSpacing);
```

Comment:
There doesn't seem to be any spacing based on m_currentConnectionsDrawCount?

Reply:
The end of DrawConnectionStatusLine() will set a new Y position so it's always ready to render the next line :)